packages/server-wallet/src/chain-service/event-tracker.ts

```diff
@@ -66,15 +66,21 @@ export class EventTracker {
     logIndex: number = Number.MAX_SAFE_INTEGER // Why default to max: getInitialHoldings can call this without logIndex,
     // in which case should be considered the latest balance in block
   ): void {
-    if (this.isNewEvent(arg.assetHolderAddress, blockNumber, logIndex)) {
+    if (this.isNewEvent(arg.asset, blockNumber, logIndex)) {
       this.managedSubscriber.holdingUpdated(arg);
     }
   }
 
   // Pass event to managed subscriber only if new
-  assetOutcomeUpdated(arg: AssetOutcomeUpdatedArg, blockNumber: number, logIndex: number): void {
-    if (this.isNewEvent(arg.assetHolderAddress, blockNumber, logIndex)) {
-      this.managedSubscriber.assetOutcomeUpdated(arg);
+  fingerprintUpdated(arg: FingerprintUpdatedArg, blockNumber: number, logIndex: number): void {
+    if (
+      this.isNewEvent(
+        arg.channelId, // TODO does this make sense to put here?
```

Comment:
Flagging for discussion.

Reply:
This file, and the chain service in general, are new to me. My shallow understanding right now looks something like:

- the `assetMap`, formerly `assetHolderMap` (line 22) is used to store the most-recent-event.
- the flagged line proposes to use the channelId as the key to index these events, since they all now originate from the same MAH/Adjudicator


Comments:
- as far as I saw, these EventTrackers [belong to individual channels](https://github.com/statechannels/statechannels/blob/f8779927900fa14838e46e11008b29b9e24da398/packages/server-wallet/src/chain-service/chain-service.ts#L62). Is this indexing redundant?
- `assetMap` probably isn't a useful name for this storage mechanism anymore?

Aside:
- the doccomments for `EventTracker` are not up-to-date

Reply:
Having come back to this without a clock ticking at me:

In the chain service each channel has its own collection of EventTrackers. The `assetMap` keys here are meant to keep separate event queues for the different potential emitters. The `channelId` serves as something distinct from the asset address keys, but so would any constant string that wasn't an asset address.

If the key here were `"fingerprint"` it might be more obvious that these fingerPrintUpdated events aren't being sorted into separate queues.

(The above assumes that this class never receives events that belong to other channels)

Reply:
This is a helpful summary. Through some manual testing I have done, it seems important that there is a separate event queue for each asset (otherwise tests fail). I renamed `assetMap` to `eventRecordFor`, and went with your suggestion for using a constant string for the fingerPrintUpdated queue (rather than the channelId).